### ISSUE TITLE
compatible with a Web Socket Server?

### ISSUE THREAD

**Issue by DetweilerRyan** [thedillonb/http-shutdown#2]
Although, I have not confirmed this with any tests, It seems to me that the shutdown method would prematurely close sockets on an http server used as the backend for a [WebSocket server](https://github.com/websockets/ws).  It seems to me that the sockets that pass through the 'connection' event and never pass through the 'request' event because they are used in the WebSocket server will always be seen as idle and will be destroyed on shutdown even if the sockets are still being used by the WebSocket server.

**Comment by thedillonb** [comment]
Yeah, I've never tried with a WS but I'd assume what you have that it probably doesn't work correctly.

**Comment by MitMaro** [comment]
I ended up having to write something to handle WebSockets: https://github.com/MitMaro/node-server-shutdown

**Comment by karlpokus** [comment]
Shouldn't a websocket be regarded similar to a keep-alive? Thus should not accept traffic post the close call?
